File: 5. Week 5 - Calendar Heatmap/solutions/python/visualization.py
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/weekly_distance_heatmap.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    df = pd.read_csv(data_path)
    
    # Pivot for Heatmap: Index=WeekDay/Year?, Columns=Week? 
    # Let's inspect columns first. Assuming Year, Week, Distance.
    # The file name implies Weekly Distance.
    # If columns are Year, Week, Distance:
    matrix = df.pivot(index='Year', columns='Week', values='Distance')
    
    plt.figure(figsize=(15, 8))
    
    # Heatmap
    sns.heatmap(matrix, cmap='Purples', linewidths=1, linecolor='white')
    
    plt.title('Weekly Distance Heatmap', fontsize=16)
    plt.xlabel('Week Number')
    plt.ylabel('Year')
    
    plt.tight_layout()
    plt.savefig('heatmap.png')
    print("Chart saved to heatmap.png")
    plt.show()

if __name__ == "__main__":
    visualize()

File: 5. Week 5 - Calendar Heatmap/solutions/r/visualization.R
# Visualize Weekly Distance Heatmap using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

data_path <- "../../data/weekly_distance_heatmap.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Plot
  p <- ggplot(df, aes(x = Week, y = Year, fill = TotalDistance)) +
    geom_tile(color = "white") +
    
    # Color Scale
    scale_fill_gradient(low = "#EDE7F6", high = "#311B92", name = "Distance (km)") +
    
    # Scale Y axis to show integer years
    scale_y_continuous(breaks = unique(df$Year)) +
    
    # Labels
    labs(
      title = "Weekly Distance Heatmap",
      x = "Week Number",
      y = "Year"
    ) +
    
    # Theme
    theme_minimal() +
    theme(
      panel.grid = element_blank()
    )
  
  ggsave("heatmap.png", plot = p, width = 12, height = 8)
  print("Chart saved to heatmap.png")

} else {
  print(paste("Error: File not found at", data_path))
}
